wire-ios-request-strategy/Sources/Helpers/FederationDeleteManagerTests.swift

```diff
@@ -0,0 +1,196 @@
+//
+
+import XCTest
+import WireDataModel
+@testable import WireRequestStrategy
+
+class FederationDeleteManagerTests: MessagingTestBase {
+    var sut: FederationDeleteManager!
+
+    override func setUp() {
+        super.setUp()
+        sut = FederationDeleteManager(syncContext: syncMOC)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        sut = nil
+    }
+
+    func testThatFederationDelete_markOneToOnConversationAsReadOnly()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            guard let conversation = otherUser.connection?.conversation else { return }
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertTrue(conversation.isForcedReadOnly)
+            XCTAssertTrue(conversation.isReadOnly)
+        }
+    }
+
+
+    func testThatFederationDelete_RemovePendingConnectionRequest()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            otherUser.connection?.status = .pending
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertEqual(otherUser.connection?.status, .ignored)
+        }
+    }
+
+
+    func testThatFederationDelete_RemoveMyConnectionRequest()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            otherUser.connection?.status = .sent
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertEqual(otherUser.connection, nil)
+        }
+    }
+
+    func testThatFederationDelete_removeSelfUserFromSecondBackendConversation()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            let conversation = createGroupConversation(with: [otherUser], hostedByDomain: domain)
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertFalse(conversation.localParticipants.contains(ZMUser.selfUser(in: syncMOC)))
+            XCTAssertTrue(conversation.localParticipants.contains(otherUser))
+        }
+    }
+
+    func testThatFederationDelete_removeOtherUserFromMyBackendConversation()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            let conversation = createGroupConversation(with: [otherUser], hostedByDomain: owningDomain)
```

Comment:
Similarly, we should assert the other user is a participant before WHEN.

Reply:
fixed